scipy/cluster/hierarchy.py

```diff
@@ -3089,10 +3089,10 @@ def dendrogram(Z, p=30, truncate_mode=None, color_threshold=None,
 
     labels : ndarray, optional
         By default, ``labels`` is None so the index of the original observation
-        is used to label the leaf nodes.  Otherwise, this is an :math:`n`
-        -sized list (or tuple). The ``labels[i]`` value is the text to put
-        under the :math:`i` th leaf node only if it corresponds to an original
-        observation and not a non-singleton cluster.
+        is used to label the leaf nodes.  Otherwise, this is an :math:`n`-sized
+        sequence, with ``n == Z.shape[1]``. The ``labels[i]`` value is the text
```

Comment:
Playing with this, it actually seems that `n == Z.shape[0] + 1`. That is how it's in the code, and I've tried the example in the dendrogram docstring with various shape of `Z`, and that seems consistent. 

Can you please verify and fix including a new test that catches this?

Reply:
The current test happened to have shape `(3, 4)`, so `Z.shape[0] + 1 == Z.shape[1]` by accident.

Reply:
Thank you so much for pointing out. You are totally right. I'm sorry. I fixed it.